plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSplitManager.java

```diff
@@ -73,9 +77,12 @@ protected ConnectorSplitSource generateSplitsForSegmentBasedScan(
         LOG.info("Got routing table for %s: %s", tableName, routingTable);
         List<ConnectorSplit> splits = new ArrayList<>();
         if (!routingTable.isEmpty()) {
-            PinotClient.TimeBoundary timeBoundary = pinotClient.getTimeBoundaryForTable(tableName);
-            generateSegmentSplits(splits, routingTable, tableName, "_REALTIME", session, timeBoundary.getOnlineTimePredicate());
-            generateSegmentSplits(splits, routingTable, tableName, "_OFFLINE", session, timeBoundary.getOfflineTimePredicate());
+            PinotClient.TimeBoundary timeBoundary = new PinotClient.TimeBoundary(null, null);
+            if (routingTable.containsKey(tableName + REALTIME_SUFFIX) && routingTable.containsKey(tableName + OFFLINE_SUFFIX)) {
+                timeBoundary = pinotClient.getTimeBoundaryForTable(tableName);
+            }
```

Comment:
This `if` should be part of next commit or the next one squashed with this. (The next commit is only adding tests - not fixing anything unlike it's title says)